services/QuillCMS/app/workers/bulk_reindex_responses_worker.rb

```diff
@@ -0,0 +1,19 @@
+require 'sidekiq/api'
+
+class BulkReindexResponsesWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: 5, queue: 'low'
+
+  def perform(starting_index, end_index, batch_size)
+    responses = Response.find_each(start: starting_index, finish: end_index, batch_size: batch_size)
```

Comment:
Nice refactor on this. So since you are using the entire batch instead of individual records, you'd want to use [`find_in_batches`](https://apidock.com/rails/ActiveRecord/Batches/find_in_batches) (it's a related method to `find_each`).

So something like this:
```ruby
Response.find_in_batches(...) do |responses|
  Response.__elasticsearch__.client.bulk(
      index: "responses",
      type: "response",
      body: responses.map { |r| {index: {_id: r.id, data: r.as_indexed_json}}}
    )
end
```

You can now use this job to loop through the responses, not just a single set, e..g. to loop through the first 100K records in chunks of 1,000, you can call:
```ruby
BulkReindexResponsesWorker.perform_async(1, 100_000, 1_000)